pkg/mpc-examples/test/01-WeightChanger.test.ts

```diff
@@ -0,0 +1,222 @@
+  const Controller = await ethers.getContractFactory(contract);
+  const c = await Controller.connect(deployer).deploy(...args);
+  const instance: Contract = await c.deployed();
+  return instance;
+}
+
+describe('WeightChanger', () => {
+  let vault: Contract;
+  let tokens: TokenList;
+
+  before('Setup', async () => {
+    ({ vault, tokens, deployer } = await setupEnvironment());
+
+    const pfpArgs = [vault.address, fp(0.1), fp(0.1)];
+    const protocolFeesProvider = await deployBalancerContract(
+      '20220725-protocol-fee-percentages-provider/',
+      'ProtocolFeePercentagesProvider',
+      deployer,
+      pfpArgs
+    );
+
+    tokenAddresses = pickTokenAddresses(tokens, 2);
+
+    const factoryTask = 'deprecated/20221021-managed-pool';
+    const libNames = ['CircuitBreakerLib', 'ManagedPoolAddRemoveTokenLib'];
+    const factoryContract = 'ManagedPoolFactory';
+    const poolFactoryArgs = [vault.address, protocolFeesProvider.address];
+    const mpFactory = await deployBalancerManagedPoolFactory(
+      factoryTask,
+      libNames,
+      factoryContract,
+      deployer,
+      poolFactoryArgs
+    );
+
+    const controllerFactoryArgs = [vault.address, mpFactory.address];
+    mpcFactory = await deployLocalContract('WeightChangerFactory', deployer, controllerFactoryArgs);
+  });
+
+  describe('Controller Deployment', () => {
+    beforeEach('deploy', async () => {
+      weightChanger = await deployController(deployer);
+    });
+
+    it("Local Controller's Vault is the Vault", async () => {
+      assert.equal(vault.address, await weightChanger.getVault());
+    });
+
+    it('Deploys managed pool; controller set as AM for all tokens', async () => {
+      const poolId = await weightChanger.getPoolId();
+      const tokens = await weightChanger.getTokens();
+
+      // Start index at 1 to skip BPT
+      for (let i = 0; i < tokens.length; i++) {
+        const info = await vault.getPoolTokenInfo(poolId, tokens[i]);
+        assert.equal(info.assetManager, weightChanger.address);
+      }
+    });
+
+    it('Initial weights are set 30/70', async () => {
+      assert.isTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), initialWeights));
+    });
+
+    it('Reweight duration set at 7 days', async () => {
+      assert.equal(await weightChanger.getReweightDuration(), time.WEEK);
+    });
+  });
+
+  // All weights are in terms of WETH/USDC
+  describe('Change Managed Pool weights', () => {
+    context('Change weights to 50/50', () => {
+      let desiredWeights: BigNumber[];
+      beforeEach('call make5050 function', async () => {
+        desiredWeights = toNormalizedWeights([fp(50), fp(50)]);
+        weightChanger = await deployController(deployer);
+        await weightChanger.make5050();
+      });
+
+      it('Failure after 4 days', async () => {
+        await fastForward(time.DAY * 4);
+        assert.isNotTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
```

Comment:
instead of just saying that the weights are wrong after 4 days, can we have a for loop verify at `n` time steps between `[t_start, t_finish]` that the weights are indeed a linear interpolation of the start/end weights?

Might require something like what João wrote in another PR that determined if values were approximately equal